### PR TITLE
EDEV-52: Media block images (and wider StreamField images)

### PR DESCRIPTION
**Pull request by jamesbiggs**
Ticket URL: [EDEV-52] + [EDEV-35]

## About these changes

This PR will bring in the ability to get useful image data from StreamFields.

I have created a new `APIImageChooserBlock` to extend `ImageChooserBlock` to change how the data is returned via the API.

Old response:
image

New response:
image

## How to check these changes

Open an article, check the API response and see that any of the images within StreamFields are now returning useful information.

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [x] Ensured that PR includes only commits relevant to the ticket
- [x] Waited for all CI jobs to pass before requesting a review
- [x] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

[EDEV-52]: https://national-archives.atlassian.net/browse/EDEV-52?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[EDEV-35]: https://national-archives.atlassian.net/browse/EDEV-35?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: etna/core/blocks/image.py [hunk 1]
```diff
@@ -9,12 +9,66 @@
...
+        self.rendition_size = rendition_size
+        super().__init__(required=required, help_text=help_text, **kwargs)
+
+    def get_api_representation(self, value, context=None):
+        jpeg_image = value.get_rendition(
+            f"{self.rendition_size}|format-jpeg|jpegquality-{self.quality}"
+        )
+        webp_image = value.get_rendition(
+            f"{self.rendition_size}|format-webp|webpquality-{self.quality}"
+        )
```
Comment (ahosgood) [inline etna/core/blocks/image.py:46]:
```suggestion
    def __init__(
        self,
        required=True,
        help_text=None,
        rendition_size="fill-600x400",
        jpegquality=60,
        webpquality=80,
        **kwargs,
    ):
        self.quality = quality
        self.rendition_size = rendition_size
        super().__init__(required=required, help_text=help_text, **kwargs)

    def get_api_representation(self, value, context=None):
        jpeg_image = value.get_rendition(
            f"{self.rendition_size}|format-jpeg|jpegquality-{self.jpegquality}"
        )
        webp_image = value.get_rendition(
            f"{self.rendition_size}|format-webp|webpquality-{self.webpquality}"
        )
```

Comment (ahosgood) [reply etna/core/blocks/image.py:46]: Could we have these separate `jpegquality` and `webpquality` parameters?

File: etna/core/blocks/image.py [hunk 2]
```diff
@@ -9,12 +9,66 @@
...
+        }
+
+
 class ImageBlock(blocks.StructBlock):
     """
     An image block which allows editors to ensure accessibility is reflected on the page.
     """
 
-    image = ImageChooserBlock(required=False)
+    image = APIImageChooserBlock(required=False)
```
Comment (ahosgood) [inline etna/core/blocks/image.py:71]: While we're here, should `required` by `true`?

File: etna/core/blocks/image.py [hunk 3]
```diff
@@ -96,7 +150,7 @@ def is_portrait(self):
 
 
 class ContentImageBlock(blocks.StructBlock):
-    image = ImageChooserBlock(required=False)
+    image = APIImageChooserBlock(required=False)
```
Comment (ahosgood) [inline etna/core/blocks/image.py:153]: As above, should `required` by `true`?

File: etna/core/blocks/image.py (hunk 2 above)
Comment (jamesbiggs) [reply etna/core/blocks/image.py:71]: Good point... Not sure why it isn't!

**Review by ahosgood — APPROVED** [review]

**Pull request merged**